layers/+lang/scheme/config.el

```diff
@@ -24,3 +24,30 @@
 ;; variables
 
 (spacemacs|define-jump-handlers scheme-mode geiser-edit-symbol-at-point)
+
+(defvar scheme-activate-chez-implementation nil
```

Comment:
Why not make them a list?

`scheme-implementation`

And use `memq` to test whether a package should be installed.

Reply:
Sounds like a good idea. I just did not have time to think about how to implement it (or look up how other layers might have implemented something like this). Will have a look a little later, no problem. Or otherwise you are welcome to edit it.

Reply:
```elisp
(defconst scheme-packages
   `(a
     b
     ,@(mapcar (lambda (x) (let ((implementations '(foo bar))))
                        (if (memq x implementations)
                            (intern (format "geiser-%s" x))
                          (user-error "Unknown scheme implmentation: %s" x)))
               scheme-implementations)))
```

Try this

Reply:
- We should use `defconst` instead of `defvar` to make sure this is immutable.
- In backquote struct, `,@` evaluate the form and splice its elements.

Reply:
Ah yes, the splice! Nice code, but unfortunately it does not work here because the `scheme-implementations` variable is not yet defined when this gets unpacked (at least when setting it in `dotspacemacs-configuration-layers`). Any suggestion how to fix this?

Reply:
Okay nice! It works, but the `scheme-implementations` have to get set in `user-init` (would be nicer if it could be set in `dotspacemacs-configuration-layers` but I think its fine) . I had not used it before, but it says [here](https://develop.spacemacs.org/doc/DOCUMENTATION.html#setting-configuration-layers-variables) that it is "a perfectly valid way to configure a layer").

Reply:
Ok, then forget about this.
Just put each of the scheme implementation in the list WITHOUT any toggle.

Then, wrap the corresponding `init` function in a test.
Without `init`, the package is defined but never downloaded.

Reply:
```elisp
(if (memq scheme-implementation scheme-implementations)
    (eval `(defun ,(intern (format "scheme/init-geiser-%s" scheme-implementation))
               (use-package ,(intern (format "geiser-%s" scheme-implementation))
                 :defer t)))
  (user-error "Unknown scheme implementation: %s" scheme-implementation))
```

Try this.

Reply:
I have found a better way. We can just set use `(memq 'implementation scheme-implementations)` for the `:toggle`. (Have updated the PR)